### PR TITLE
Update ember-data to 3.28

### PR DESCRIPTION
**Pull request by elpoelma**
This PR includes an update to ember-data 3.28.12. Ember-data 3.28 includes some changes to relationships which broke some of the functionality of the app. This PR thus additionally includes changes to the models and components.

- Update to ember data 3.28.12
- Add polymorphic attribute to necessary models
- Move constructor contents to didInsert hooks so relationships can be properly awaited.

### CODE REVIEW

File: tests/unit/models/code-list-test.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { module, test } from 'qunit';
```
Comment (lagartoverde) [inline tests/unit/models/code-list-test.js:1]: Do we want this test?

**Review by lagartoverde — APPROVED** [review]
Looks good

File: tests/unit/models/code-list-test.js (hunk 1 above)
Comment (elpoelma) [reply tests/unit/models/code-list-test.js:1]: I added it in order to more easily test the hasMany issue that was solved with the polymorphic attribute. I don't know if we should keep it.

Comment (lagartoverde) [reply tests/unit/models/code-list-test.js:1]: Ok, then we keep it :), you can merge

**Pull request merged**